jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java

```diff
@@ -102,6 +105,38 @@
         appRoot, rawConfiguration.getJvmFlags(), mainClass);
   }
 
+  /**
+   * Checks whether or not current Java version is compatible with the base image.
+   *
+   * @param baseImageConfiguration the configured base image
+   * @param fromImageParameterName the name of the base image configuration parameter
+   * @throws InvalidImageReferenceException if the project is incompatible with the base image
+   */
+  public static void checkJavaVersion(
+      @Nullable String baseImageConfiguration, String fromImageParameterName)
+      throws InvalidImageReferenceException {
+    // TODO: check for other Java 8 base images?
+    if (baseImageConfiguration == null
+        || baseImageConfiguration.equals(DEFAULT_BASE_IMAGE)
+        || baseImageConfiguration.equals(DEFAULT_BASE_IMAGE_WAR)) {
+      String version = System.getProperty("java.version");
```

Comment:
I think this reads the JVM version running Maven/Gradle, which may do the job. I just wonder if we should be reading the target version in the build files. For example, looking at `<maven.compiler.target>` or `<maven.complier.release>` (this property seems new after Java 8) or the `<target>` version of `maven-compiler-plugin`. @GoogleContainerTools/java-tools thoughts?

Reply:
Yea I agree that this should use the target version defined in the build file since that could be completely different from the JVM version used to run Maven/Gradle.

Reply:
Any idea how to do this?

Reply:
It looks like in Maven, this can be determined from:

1. `maven-compiler-plugin` configuration `<release>` or `<target>` or `<compilerVersion>`
1. properties `maven.compiler.release` or `maven.compiler.target`
1. could potentially be inherited from parent pom - could recursively check parent poms or perhaps look at the effective pom (like what `mvn help:effective-pom` shows)

In gradle, I think it's more straightforward - just look at `targetCompatibility` from the `compileJava` extension.

Reply:
Another alternative that might be easier is to use javassist to read a class file and check its major version.